verilog/formatting/token_annotator_test.cc

```diff
@@ -1345,6 +1345,33 @@ TEST(TokenAnnotatorTest, AnnotateFormattingWithContextTest) {
           {0, SpacingOptions::MustAppend},
       },
 
+      // Handle '&' as binary
+      {
+          DefaultStyle,
+          {'&', "&"},
+          {'~', "~"},
+          {},
+          {1, SpacingOptions::Undecided},
+      },
+
+      // Handle '&' as unary
+      {
+          DefaultStyle,
+          {'&', "&"},
+          {yytokentype::TK_DecNumber, "42"},
+          {NodeEnum::kUnaryPrefixExpression},
+          {0, SpacingOptions::MustAppend},
+      },
+
+      // Two unary operators
+      {
+          DefaultStyle,
+          {'~', "~"},
+          {'~', "~"},
+          {NodeEnum::kUnaryPrefixExpression},
+          {0, SpacingOptions::MustAppend},
+      },
+
```

Comment:
few more test case suggestions:
* '~' '('
* '~' SymbolIdentifier

Reply:
Sure, that makes sense, Added the  test for both ~(, ~Identifier, and &(, &identifier